src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenManagementService.cs

```diff
@@ -59,4 +65,38 @@ private async Task<ClientCredentialsToken> SynchronizeAsync(string clientName, F
             this._lazyGetNewTokenTasks.TryRemove(clientName, out _);
         }
     }
+
+    private Func<Task<ClientCredentialsToken>> GetNewTokenTaskFactory(string clientName, CancellationToken cancellationToken) => async () =>
+    {
+        var newToken = await this._tokenEndpointService.RequestTokenAsync(clientName, cancellationToken).ConfigureAwait(false);
```

Comment:
If this call fails (exception), I think the background refresh loop is stopped as you will never call `ScheduleTokenBackgroundRefreshAsync` again. Is it what we want?

Reply:
Actually this is not true :)

If the background refresh loop is stopped because we failed to retrieve a token, then the next time an authenticated call is made and goes through our HttpClient delegated handlers, we will request a new token, then start the background task again => starting the loop.

I did QA that by shutting down my identity server. Token refresh was set to every 5 seconds. The background loop stopped. Started my identity server again, made an authenticated call, and background refresh loop started again.